instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt

```diff
@@ -160,6 +160,9 @@ class LoggerBuilderE2ETests {
         }
         val viewKey = forge.anAlphabeticalString()
         GlobalRum.get().startView(viewKey, forge.anAlphabeticalString())
+        InstrumentationRegistry.getInstrumentation().waitForIdleSync()
+        // Give time to the View event to be propagated
+        Thread.sleep(200)
         logger.sendRandomLog(testMethodName, forge)
         GlobalRum.get().stopView(viewKey)
```

Comment:
otherwise by the time `stopView` is called there is a chance that there is no view registered yet?

Reply:
No it is just that when sending the log it will try to get the `RUM` context and because that `View` was not yet recorded the `view.id` will be missing from the context.

Reply:
hm, then I would say it is better to solve it in the production code, because this is the issue user can experience, but I guess solution is not that easy. we maybe can create an issue and come back to it later? wdyt?

Reply:
Well this will mean synchronising Logs and RUM, not sure if we want that but we can discuss during the daily.

Reply:
We have similar issue on iOS and we don't address it on purpose - synchronising features would be very expensive and not sure if worth it. In most cases, instrumentation code is spread between different parts and threads of the application (e.g. RUM auto instrumentation running "transparently" and logger being called explicitly) so this case is not that user-facing IMHO.

On the other hand, we do synchronize user-facing configuration APIs, e.g. calling `setTrackingConsent(.notGranted)` would be applied immediately to any next data send from the same thread, same as `.setUserInfo(id:email:name)`.

Reply:
Yes, that's usually our case too, we do synchronize these public APIs at the feature level.